src/pages/Animetoast/main.ts

```diff
@@ -0,0 +1,82 @@
+import { pageInterface } from "../pageInterface";
+import { SyncPage } from "../syncPage";
+
+export const Animetoast: pageInterface = {
+    name: "Animetoast",
+    domain: "https://www.animetoast.cc",
+    languages: ['German'],
+    type: "anime",
+    isSyncPage: function (url: string): boolean {
+        return $('.entry-title.light-title').text() != null && url.includes('?link=');
+    },
+    isOverviewPage: function (url: string): boolean {
+        return $('.entry-title.light-title').text() != null && !url.includes('?link=');
+    },
+    sync: {
+        getTitle: function (url: string): string {
+            let title = $('.entry-title.light-title').text().split(' ');
+            if(!(title.length >= 2))return title.join(' ');
+            for(let i = title.length - 1; i >= title.length - 2; i--){
+                const t = title[i];
+                t.match(/(Ger|Sub|Dub)/gi) ? title.splice(i, 1) : null;
+            }
+            return title.join(' ');
+        },
+        getIdentifier: function (url: string): string {
+            let title = $('.entry-title.light-title').text().split(' ');
+            if(!(title.length >= 2))return title.join(' ');
+            for(let i = title.length - 1; i >= title.length - 2; i--){
+                const t = title[i];
+                t.match(/(Ger|Sub|Dub)/gi) ? title.splice(i, 1) : null;
+            }
+            return title.join(' ');
+        },
+        getOverviewUrl: function (url: string): string {
+            return url.split('?link=')[0];
+        },
+        getEpisode: function (url: string): number {
+            let s = $('.current-link').text().match(/(\d+)/gi);
+            return Number((s![s!?.length-1]));
```

Comment:
I think you can remove the `g` from the regex flags then the index of the array stays the same.
using ! is here the wrong way. Better check if s is null and return a NaN. And better use real variable names not only s.
```
if(!s) return NaN;
return Number(s[2]);
```